test/libs/helper-functions.js

```diff
@@ -0,0 +1,194 @@
+/**
+ * Copyright 2015 Google Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+/* eslint-env browser */
+
+// Each service worker that is registered should be given a unique
+// scope. To achieve this we register it with a scope the same as
+// an iframe's src that is unique for each test.
+// Service workers will then be made to claim pages on this scope -
+// i.e. the iframe
+var getIframe = function() {
+  return new Promise(resolve => {
+    var existingIframe = document.querySelector('.js-test-iframe');
+    if (existingIframe) {
+      return resolve(existingIframe);
+    }
+
+    var newIframe = document.createElement('iframe');
+    newIframe.classList.add('js-test-iframe');
+    newIframe.src = '/test/iframe/' + Math.random();
```

Comment:
Could you use a globally-scoped `iframeId` counter that gets incremented after each iframe is created? If you don't think that's sufficient because of the caching concerns in between runs, then combine the counter with `Date.now()`.

Including `Math.random()` in a test introduces an infinitesimal chance of unnecessary flakiness.